km52/Antiushyn_Yaroslav/laboratory1.sql

```diff
@@ -1,2 +1,136 @@
--- LABORATORY WORK 1
--- BY Antiushyn_Yaroslav
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+вставляти дані до таблиць
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+/*
+CREATE User Antiyshin IDENTIFIED by Antiyshin
+DEFAULT TABLESPACE "USERS"
+TEMPORARY TABLESPACE "TEMP";
+ALTER USER Antiyshin QUOTA 100M ON USERS:
+GRANT "CONNECT" to Antiyshin;
+INSERT INTO to USERS; 
+
+*/
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Класна кімната має парти та стільці.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+CREATE User Antiyshin IDENTIFIED by Antiyshin
+DEFAULT TABLESPACE "USERS"
+TEMPORARY TABLESPACE "TEMP";
+ALTER USER Antiyshin QUOTA 100M ON USERS:
+
+CREATE TABLE Desk (desk_id varchar(30) not null);
+ALTER TABLE Desk 
+  ADD CONSTRAINT desk_id_pk PRIMARY KEY (desk_id);
+  
+CREATE TABLE Chair (chair_id varchar(30) not null);
+ALTER TABLE Desk 
+  ADD CONSTRAINT chair_id_pk  PRIMARY KEY (chair_id);
+
+CREATE TABLE Classroom (
+  desk_id_fk varchar(30) not null
+  chair_id_fk varchar(30) not null
+);
+
+ALTER TABLE Classroom 
+  ADD CONSTRAINT desk_id_pk PRIMARY KEY (desk_id_fk,chair_id_fk);
+  
+Alter
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+  
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+3.a. 
+Скільком покупцям продано найдорожчий товар?
+Виконати завдання в SQL. 
+4 бали
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
+
```

Comment:
```
COUNT (
    PROJECT(customers WHERE price = MAX(price)){price}
)
```

Reply:
Select count(cust_id)
From ( Select Distinct Customers.cust_id, Products.prod_price
From Customers, Orders, Orderitems, Products
Where Customers.cust_id = Orders.cust_id
and Orders.order_num = Orderitems.order_num
and Orderitems.prod_id = Products.prod_id
and Products.prod_price = max(prod_price))